2-Draft-Accepted/RFCxxxx-PowerShellGet-3.0.md

```diff
@@ -0,0 +1,348 @@
+a hash table where `repository` is set to the URL of the repository and
+`version` contains the [Nuget version range syntax](https://docs.microsoft.com/en-us/nuget/reference/package-versioning#version-ranges-and-wildcards).
+
+```powershell
+Install-PSResource -RequiredResources @{
+  "Configuration" = "[1.3.1,2.0)"
+  "Pester"        = @{
+    version = "[4.4.2,4.7.0]"
+    repository = "https://www.powershellgallery.com"
+    credential = $cred
+  }
+}
+```
+
+The json format will be the same as if this hashtable is passed to `ConvertTo-Json`:
+
+```json
+{
+  "Pester": {
+    "version": "[4.4.2,4.7.0]",
+    "credential": null,
+    "repository": "https://www.powershellgallery.com"
+  },
+  "Configuration": "[1.3.1,2.0)"
+}
+```
+
+>[!NOTE]
+>Credentials management will be using an upcoming RFC for generalized
+>credential management.  Not sure how to make that work in json other
+>than requiring json to have credential as a json object but that means
+>password/secret is in clear text.
+
+The `repository` property can be the name of the registered repository or the URL
+to a repository.
+If no `repository` is specified, it will use the `default` repository.
+
+The older `System.Version` four part version type will be supported to retain
+compatibility with existing published modules using that format.
+
+Due to the differences between semver and `System.Version`, we may have to keep
+the two distinct.
+
+Dependent declared modules not found on the system will prompt to be installed
+including module name, version, and size unless `-IncludeDependencies` is
+specified which will install without prompting.
+Declared dependencies are only searched within the same repository as the original
+module to be installed.
+
+### Saving resources
+
+This cmdlet uses `Install-PSResource -DestinationPath`.
+
+With the removal of PackageManagement, there is still a need to support saving
+arbitrary nupkgs (assemblies) used for scripts.
+
+`Save-PSResource -Type Library` will download nupkgs that have a `lib` folder.
+The dependent native library in `runtimes` matching the current system runtime
+will be copied to the root of the destination specified.
+A `-IncludeAllRuntimes` can be used to explicitly retain the `runtimes` directory
+hierarchy within the nupkg to the root of the destination.
+
+>[!NOTE]
+>Library support may not be available in 3.0.
+
+A `-Prerelease` switch allows saving prerelease versions.
+If the `-Version` includes a prerelease label like `2.0.0-beta4`, then this
+switch is not necessary and the prerelease version will be installed.
+
+### Updating resources
+
+`Update-PSResource` will update all resources to most recent minor version by default.
+A `-AllowMajorVersionUpdate` switch will allow updating to newer major version.
+A `-PatchUpdateOnly` switch will allow updating only to patched versions.
+
+If the installed resource is a pre-release, it will automatically update to
+latest prerelease or stable version (if available).
```

Comment:
For a stable version, is there a way to update it to the latest prerelease?

Reply:
The `-Prerelease` switch is for that purpose